### PR TITLE
add optional "storedValues" attribute for persistent indexes

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

The feature is described in the docs PR in detail, so there is no separate PR description here.
Docs PR: https://github.com/arangodb/docs/pull/858

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [x] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [x] :book: CHANGELOG entry made

#### Related Information

- [x] Docs PR: https://github.com/arangodb/docs/pull/858

### Testing & Verification

- [x] This PR adds tests that were used to verify all changes:
  - [x] Added new **integration tests** (shell_client, shell_server)

### CODE REVIEW

**Comment by gnusi** [comment]
In ArangoSearch we have similar concept called "storedValues", would be nice to use similar term

**Comment by jsteemann** [comment]
> In ArangoSearch we have similar concept called "storedValues", would be nice to use similar term

The index lookup values (a.k.a. "fields") are also stored in the index.
I don't think "storedValues" is a good term, because it someone implies that the other values (in "fields") are _not_ stored. The same is true for arangosearch by the way.
Maybe there is even a better term... Anyway, it is just naming, and changing it later (before merging) will be easy.

**Comment by gnusi** [comment]
@jsteemann naming is also important for `inverted` index, currently we use term `storedValues` similar to views. So we have to agree on naming to be consistent.

**Comment by Dronplane** [comment]
As a suggestion: we have a PR https://github.com/arangodb/arangodb/pull/14240 I hope to be merged soon. This PR introduces a `CoveringData` abstraction for the index callbacks. Could we wait for #14240 to be merged and use the same abstraction here? So callback will not "know" about index projections  storage schemes.

**Comment by jsteemann** [comment]
Setting this to draft state until merge of #14240.

File: arangod/RocksDBEngine/RocksDBVPackIndex.cpp
Comment (Dronplane) [inline arangod/RocksDBEngine/RocksDBVPackIndex.cpp:944]: What do we want to in case of missing attribute? For the `fields` we either stop indexing document (sparse index)  or forcefully put `null` slice. As I see here we may put `none`. Maybe always force `null` for consistency with `fields`?

**Review by Dronplane — APPROVED** [review]
LGTM. Left couple comments regarding missing `storedValues` attributes in  the indexed document

File: arangod/RocksDBEngine/RocksDBVPackIndex.cpp
Comment (jsteemann) [reply arangod/RocksDBEngine/RocksDBVPackIndex.cpp]: Addressed.

File: arangod/RocksDBEngine/RocksDBVPackIndex.cpp
Comment (jsteemann) [reply arangod/RocksDBEngine/RocksDBVPackIndex.cpp]: Addressed.

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**
